src/app.ts

```diff
@@ -13,14 +14,61 @@ interface TemperatureSummary {
   average: number
 }
 
+interface CustomTemperatureSummary extends TemperatureSummary {
+  count: number
+}
+
+const globalMap: Map<string, Map<number, CustomTemperatureSummary>> = new Map()
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  for (const { time, temperature, city } of readings) {
+    if (!globalMap.has(city)) {
+      globalMap.set(city, new Map())
+    }
+
+    const cityMap: Map<number, CustomTemperatureSummary> = globalMap.get(
+      city,
+    ) as Map<number, CustomTemperatureSummary>
+    const numberTime: number = time.getTime()
```

Comment:
I was reviewing the documentation and some forums, and it is generally recommended to use has in cases where the key might be falsy. Since this doesn’t seem to be the case here, I agree that we can rely on using get.  I've improved the code in this commit: https://github.com/ravnhq/TS-Nederdy-Challenges/pull/52/commits/e8251c1c3d0b1fe3864ccf8eb7be4a91f3eb5607